receiver/sapmreceiver/trace_receiver_test.go

```diff
@@ -0,0 +1,285 @@
+		},
+		Spans: []*model.Span{
+			{
+				TraceID:       traceID,
+				SpanID:        childSpanID,
+				OperationName: "DBSearch",
+				StartTime:     t1,
+				Duration:      d1,
+				Tags: []model.KeyValue{
+					model.String(tracetranslator.TagStatusMsg, "Stale indices"),
+					model.Int64(tracetranslator.TagStatusCode, trace.StatusCodeNotFound),
+					model.Bool("error", true),
+				},
+				References: []model.SpanRef{
+					{
+						TraceID: traceID,
+						SpanID:  parentSpanID,
+						RefType: model.SpanRefType_CHILD_OF,
+					},
+				},
+			},
+			{
+				TraceID:       traceID,
+				SpanID:        parentSpanID,
+				OperationName: "ProxyFetch",
+				StartTime:     t1.Add(d1),
+				Duration:      d2,
+				Tags: []model.KeyValue{
+					model.String(tracetranslator.TagStatusMsg, "Frontend crash"),
+					model.Int64(tracetranslator.TagStatusCode, trace.StatusCodeInternal),
+					model.Bool("error", true),
+				},
+			},
+		},
+	}
+}
+
+// sendSapm acts as a client for sending sapm to the receiver.  This could be replaced with a sapm exporter in the future.
+func sendSapm(endpoint string, sapm *splunk_sapm.PostSpansRequest, zipped bool) (*http.Response, error) {
+	// marshal the sapm
+	reqBytes, err := proto.Marshal(sapm)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal sapm %v", err.Error())
+	}
+
+	if zipped {
+		// create a gzip writer
+		var buff bytes.Buffer
+		writer := gzip.NewWriter(&buff)
+
+		// run the request bytes through the gzip writer
+		_, err = writer.Write(reqBytes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to write gzip sapm %v", err.Error())
+		}
+
+		// close the writer
+		err = writer.Close()
+		if err != nil {
+			return nil, fmt.Errorf("failed to close the gzip writer %v", err.Error())
+		}
+
+		// save the gzipped bytes as the request bytes
+		reqBytes = buff.Bytes()
+	}
+
+	// build the request
+	req, _ := http.NewRequest(http.MethodPost, fmt.Sprintf("http://%s%s", endpoint, TraceEndpointV2), bytes.NewReader(reqBytes))
+	req.Header.Set("Content-Type", "application/x-protobuf")
+
+	// set headers for gzip
+	if zipped {
+		req.Header.Set("Content-Encoding", "gzip")
+		req.Header.Set("Accept-Encoding", "gizp")
+	}
+
+	// send the request
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return resp, fmt.Errorf("failed to send request to receiver %v", resp)
+	}
+
+	return resp, nil
+}
+
+func TestReception(t *testing.T) {
+	now := time.Unix(1542158650, 536343000).UTC()
+	nowPlus10min := now.Add(10 * time.Minute)
+	nowPlus10min2sec := now.Add(10 * time.Minute).Add(2 * time.Second)
+
+	type args struct {
+		config *Config
+		sapm   *splunk_sapm.PostSpansRequest
+		zipped bool
+	}
+	tests := []struct {
+		name string
+		args args
+		want []consumerdata.TraceData
+	}{
+		{
+			name: "receive uncompressed sapm",
+			args: args{
+				// 1. Create the SAPM receiver aka "server"
+				config: &Config{
+					ReceiverSettings: configmodels.ReceiverSettings{Endpoint: ":7276"},
```

Comment:
I can't use that because that's in a package internal to `optentelemetry-collector` and isn't in `opentelemetry-collector-contrib`.  But since this is the default endpoint, i'll use the constant in this package.

Reply:
I will take care of exposing that soon (no need to hold this PR for that).